### ISSUE TITLE
Pod meta info

### ISSUE THREAD

**Issue by sladage** [qt-pods/qt-pods#22]
Currently it seems the meta information for the various pods is managed manually by you? It might be better if the meta information is retreived from the git repositiories of each pod.

**Comment by jacob3141** [comment]
True. The only thing that kept me from doing that was that you need to clone a git repository as a whole in order to access a single file (which would heavily decrease performance). Suggestions on how to do that better are welcome!

**Comment by sladage** [comment]
If the repository is on github you could get the content of a single file without cloning. Not sure about other git hosts.

**Comment by jacob3141** [comment]
Yes, you are right. We would need to force people to use a git host that supports that, plus we would need to perform an extra request for each pod (which can take very long dependning on the speed of the servers hosting the projects). I really think it's best in terms of speed to keep all meta data in one place, although I agree with you that architecturally, it would be best to keep all meta data in one place.

Also be aware that anyone can set up a source information file (pods.json) on his own server. So, for example, If you were a developer maintaining a handful of projects you can offer your own metafile for your projects and people can add it as a source to their qtpods.

**Comment by sladage** [comment]
Since you want to keep things centralized, you could also collect the meta data from each repositiory server-side. Say once a day or so. Or you could create a way for maintainers to push meta data to your server.

**Comment by jacob3141** [comment]
Ok, I leave this ticket as a reminder for implemented such a feature. If you or anyone else wants to come up with a concrete implementation, we can discuss and I will gladly pull it in.
